typed/parser.go

```diff
@@ -29,7 +29,7 @@ type YAMLObject string
 
 // Parser implements YAMLParser and allows introspecting the schema.
 type Parser struct {
-	Schema schema.Schema
+	Schema *schema.Schema
```

Comment:
Where do we create that parser?

Reply:
This library provides one constructor here which uses YAML unmarshalling to function, so it does not appear to require modification:

https://github.com/kubernetes-sigs/structured-merge-diff/blob/73412afd729b9cdaa8d4f87341b04e2877cdff97/typed/parser.go#L53


None of the other usages are in this library, so they will have to be updated separately when they update their dependencies. Here are a few I found:

https://github.com/kubernetes/apimachinery/blob/master/pkg/util/managedfields/gvkparser.go#L62
https://github.com/kubernetes/client-go/blob/28ccde769fc5519dd84e5512ebf303ac86ef9d7c/util/jsonpath/parser.go#L63

Reply:
Sounds good, that second link is unrelated though.